{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import glob\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Confiugre IDseq Reports to match format of other data\n",
    "\n",
    "1. Download the IDseq reports files from the UI for project = Cell Reports Benchmarking Metagenomic Tools\n",
    "2. run this script to reformat the results into the format for the Cell paper benchmarking notebook.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['UnAmbiguouslyMapped_ds.soil',\n",
       " 'UnAmbiguouslyMapped_ds.buccal',\n",
       " 'UnAmbiguouslyMapped_ds.cityparks',\n",
       " 'UnAmbiguouslyMapped_ds.gut',\n",
       " 'UnAmbiguouslyMapped_ds.hous1',\n",
       " 'UnAmbiguouslyMapped_ds.7',\n",
       " 'UnAmbiguouslyMapped_ds.hous2',\n",
       " 'UnAmbiguouslyMapped_ds.nycsm',\n",
       " 'atcc_even',\n",
       " 'atcc_staggered']"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "idseq_reports_dir = './data/idseq/cell-reports-benchmarking-metagenomics-tools_reports_v3.13/'\n",
    "\n",
    "filename_list = glob.glob(idseq_reports_dir + '*.csv')\n",
    "sample_names = [i.split('/')[-1].split('.')[0].replace('-','.') for i in filename_list]\n",
    "sample_names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "./data/idseq/cell-reports-benchmarking-metagenomics-tools_reports_v3.13/UnAmbiguouslyMapped_ds-soil.csv\n",
      "UnAmbiguouslyMapped_ds.soil\n",
      "./data/idseq/cell-reports-benchmarking-metagenomics-tools_reports_v3.13/UnAmbiguouslyMapped_ds-buccal.csv\n",
      "UnAmbiguouslyMapped_ds.buccal\n",
      "./data/idseq/cell-reports-benchmarking-metagenomics-tools_reports_v3.13/UnAmbiguouslyMapped_ds-cityparks.csv\n",
      "UnAmbiguouslyMapped_ds.cityparks\n",
      "./data/idseq/cell-reports-benchmarking-metagenomics-tools_reports_v3.13/UnAmbiguouslyMapped_ds-gut.csv\n",
      "UnAmbiguouslyMapped_ds.gut\n",
      "./data/idseq/cell-reports-benchmarking-metagenomics-tools_reports_v3.13/UnAmbiguouslyMapped_ds-hous1.csv\n",
      "UnAmbiguouslyMapped_ds.hous1\n",
      "./data/idseq/cell-reports-benchmarking-metagenomics-tools_reports_v3.13/UnAmbiguouslyMapped_ds-7.csv\n",
      "UnAmbiguouslyMapped_ds.7\n",
      "./data/idseq/cell-reports-benchmarking-metagenomics-tools_reports_v3.13/UnAmbiguouslyMapped_ds-hous2.csv\n",
      "UnAmbiguouslyMapped_ds.hous2\n",
      "./data/idseq/cell-reports-benchmarking-metagenomics-tools_reports_v3.13/UnAmbiguouslyMapped_ds-nycsm.csv\n",
      "UnAmbiguouslyMapped_ds.nycsm\n",
      "./data/idseq/cell-reports-benchmarking-metagenomics-tools_reports_v3.13/atcc_even.csv\n",
      "atcc_even\n",
      "./data/idseq/cell-reports-benchmarking-metagenomics-tools_reports_v3.13/atcc_staggered.csv\n",
      "atcc_staggered\n"
     ]
    }
   ],
   "source": [
    "database = 'default'\n",
    "ntnr_dict = {'nt':'NT_r', 'nr':'NR_r'}\n",
    "\n",
    "list_of_result_dfs = []\n",
    "\n",
    "for filename in filename_list:\n",
    "    \n",
    "    print(filename)\n",
    "    \n",
    "    sample_name = filename.split('/')[-1].split('.')[0].replace('-','.')\n",
    "    print(sample_name)\n",
    "    \n",
    "    file_df = pd.read_csv(filename)\n",
    "    \n",
    "    for j in ['nt','nr', 'ntnr_conc', 'ntnr_conc_stringent']:\n",
    "        classifier = 'idseq_' + j\n",
    "        \n",
    "        metric = j\n",
    "        if j in ['ntnr_conc','ntnr_conc_stringent']: # for concordance, use NT values when NR r > 0\n",
    "            metric = 'nt'\n",
    "        \n",
    "        # select only the species-level taxa\n",
    "        species = file_df[file_df.tax_level == 1]\n",
    "        \n",
    "        # keep species with r > 0\n",
    "        species = species[species[ntnr_dict[metric]] > 0]\n",
    "        \n",
    "        if j == 'ntnr_conc': # require NT and NR be greater than zero\n",
    "            species = species[species['NT_r'] > 0]\n",
    "            species = species[species['NR_r'] > 0]\n",
    "            \n",
    "        if j == 'ntnr_conc_stringent':\n",
    "            species = species[species['NT_rpm'] > 10]\n",
    "            species = species[species['NR_rpm'] > 10]\n",
    "            \n",
    "        # take a subset of the columns\n",
    "        species = species[['tax_id','name',ntnr_dict[metric]]]\n",
    "        \n",
    "        # convert raw r values to proportion of total\n",
    "        species['cum_abundance'] = species[ntnr_dict[metric]] / species[ntnr_dict[metric]].sum()\n",
    "        \n",
    "        # append fields required by Ye et al script\n",
    "        species['classrank'] = ['species' for i in range(len(species.index))]\n",
    "        species['rank'] = ['species' for i in range(len(species.index))]\n",
    "        species['classifier'] = [classifier for i in range(len(species.index))]\n",
    "        species['database'] = [database for i in range(len(species.index))]\n",
    "        species['sample'] = sample_name\n",
    "        species['cls_cum_abundance'] = [None for i in range(len(species.index))]\n",
    "        species['unique_abundance'] = [None for i in range(len(species.index))]\n",
    "        species['cls_unique_abundance'] = [None for i in range(len(species.index))]\n",
    "\n",
    "        species = species[['sample', 'classifier','database','tax_id','name','cum_abundance','cls_cum_abundance',\n",
    "                      'unique_abundance','cls_unique_abundance','classrank','rank']]  \n",
    "\n",
    "\n",
    "        # select only teh genus-level taxa\n",
    "        genus = file_df[file_df.tax_level == 2]#[['tax_id','name',ntnr_dict[metric]]]\n",
    "        #genus.columns = ['taxid','name',ntnr_dict[j]]\n",
    "        \n",
    "        # keep genera with r > 0\n",
    "        genus = genus[genus[ntnr_dict[metric]] > 0]\n",
    "        \n",
    "        if j == 'ntnr_conc': # require NT and NR be greater than zero\n",
    "            genus = genus[genus['NT_r'] > 0]\n",
    "            genus = genus[genus['NR_r'] > 0]\n",
    "            \n",
    "        if j == 'ntnr_conc_stringent':\n",
    "            genus = genus[genus['NT_rpm'] > 10]\n",
    "            genus = genus[genus['NR_rpm'] > 10]\n",
    "            \n",
    "        # take a subset of the columns\n",
    "        genus = genus[['tax_id','name',ntnr_dict[metric]]]\n",
    "        \n",
    "        \n",
    "        # convert raw r values to proportion of total\n",
    "        genus['cum_abundance'] = genus[ntnr_dict[metric]] / genus[ntnr_dict[metric]].sum()\n",
    "        genus['classrank'] = ['genus' for i in range(len(genus.index))]\n",
    "        genus['rank'] = ['genus' for i in range(len(genus.index))]\n",
    "        genus['classifier'] = [classifier for i in range(len(genus.index))]\n",
    "        genus['database'] = [database for i in range(len(genus.index))]\n",
    "        genus['samplename'] = sample_name\n",
    "        genus['sample'] = sample_name\n",
    "        genus['cls_cum_abundance'] = [None for i in range(len(genus.index))]\n",
    "        genus['unique_abundance'] = [None for i in range(len(genus.index))]\n",
    "        genus['cls_unique_abundance'] = [None for i in range(len(genus.index))]\n",
    "\n",
    "        genus = genus[['sample', 'classifier','database','tax_id','name','cum_abundance','cls_cum_abundance',\n",
    "                      'unique_abundance','cls_unique_abundance','classrank','rank']]        \n",
    "    \n",
    "        list_of_result_dfs.append(species)\n",
    "        list_of_result_dfs.append(genus)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# output the results to file: idseq_combined_results.tsv\n",
    "final_results = pd.concat(list_of_result_dfs)\n",
    "final_results.columns = ['sample','classifier','database','taxid','name','cum_abundance','cls_cum_abundance','unique_abundance','cls_unique_abundance','classrank','rank']\n",
    "final_results.to_csv('./data/idseq/cell-reports-benchmarking-metagenomics-tools_reports_v3.13/idseq_combined_results.tsv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>classifier</th>\n",
       "      <th>database</th>\n",
       "      <th>taxid</th>\n",
       "      <th>name</th>\n",
       "      <th>cum_abundance</th>\n",
       "      <th>cls_cum_abundance</th>\n",
       "      <th>unique_abundance</th>\n",
       "      <th>cls_unique_abundance</th>\n",
       "      <th>classrank</th>\n",
       "      <th>rank</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>UnAmbiguouslyMapped_ds.soil</td>\n",
       "      <td>idseq_nt</td>\n",
       "      <td>default</td>\n",
       "      <td>940615</td>\n",
       "      <td>Granulicella tundricola</td>\n",
       "      <td>0.019947</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>species</td>\n",
       "      <td>species</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>UnAmbiguouslyMapped_ds.soil</td>\n",
       "      <td>idseq_nt</td>\n",
       "      <td>default</td>\n",
       "      <td>940614</td>\n",
       "      <td>Granulicella mallensis</td>\n",
       "      <td>0.020603</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>species</td>\n",
       "      <td>species</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>UnAmbiguouslyMapped_ds.soil</td>\n",
       "      <td>idseq_nt</td>\n",
       "      <td>default</td>\n",
       "      <td>210</td>\n",
       "      <td>Helicobacter pylori</td>\n",
       "      <td>0.020289</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>species</td>\n",
       "      <td>species</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>UnAmbiguouslyMapped_ds.soil</td>\n",
       "      <td>idseq_nt</td>\n",
       "      <td>default</td>\n",
       "      <td>138563</td>\n",
       "      <td>Helicobacter cetorum</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>species</td>\n",
       "      <td>species</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>UnAmbiguouslyMapped_ds.soil</td>\n",
       "      <td>idseq_nt</td>\n",
       "      <td>default</td>\n",
       "      <td>96345</td>\n",
       "      <td>Flavobacterium psychrophilum</td>\n",
       "      <td>0.019518</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>species</td>\n",
       "      <td>species</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         sample classifier database   taxid  \\\n",
       "1   UnAmbiguouslyMapped_ds.soil   idseq_nt  default  940615   \n",
       "2   UnAmbiguouslyMapped_ds.soil   idseq_nt  default  940614   \n",
       "9   UnAmbiguouslyMapped_ds.soil   idseq_nt  default     210   \n",
       "10  UnAmbiguouslyMapped_ds.soil   idseq_nt  default  138563   \n",
       "15  UnAmbiguouslyMapped_ds.soil   idseq_nt  default   96345   \n",
       "\n",
       "                            name  cum_abundance cls_cum_abundance  \\\n",
       "1        Granulicella tundricola       0.019947              None   \n",
       "2         Granulicella mallensis       0.020603              None   \n",
       "9            Helicobacter pylori       0.020289              None   \n",
       "10          Helicobacter cetorum       0.000001              None   \n",
       "15  Flavobacterium psychrophilum       0.019518              None   \n",
       "\n",
       "   unique_abundance cls_unique_abundance classrank     rank  \n",
       "1              None                 None   species  species  \n",
       "2              None                 None   species  species  \n",
       "9              None                 None   species  species  \n",
       "10             None                 None   species  species  \n",
       "15             None                 None   species  species  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_results.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'idseq_nr', 'idseq_nt', 'idseq_ntnr_conc', 'idseq_ntnr_conc_stringent'}"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(final_results['classifier'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "sample_name\n",
      "atcc_even                           0.849691\n",
      "atcc_staggered                      0.837371\n",
      "RL_S001                             0.825973\n",
      "RM2_S001                            0.772459\n",
      "RM2_S002                            0.823163\n",
      "UnAmbiguouslyMapped_ds.7            0.852747\n",
      "UnAmbiguouslyMapped_ds.buccal       0.958213\n",
      "UnAmbiguouslyMapped_ds.cityparks    0.985357\n",
      "UnAmbiguouslyMapped_ds.gut          0.973906\n",
      "UnAmbiguouslyMapped_ds.hous1        0.980988\n",
      "UnAmbiguouslyMapped_ds.hous2        0.979932\n",
      "UnAmbiguouslyMapped_ds.nycsm        0.977828\n",
      "UnAmbiguouslyMapped_ds.soil         0.983160\n",
      "dtype: float64\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>total_reads</th>\n",
       "      <th>reads_after_star</th>\n",
       "      <th>reads_after_trimmomatic</th>\n",
       "      <th>reads_after_priceseq</th>\n",
       "      <th>reads_after_cdhitdup</th>\n",
       "      <th>nonhost_reads</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>atcc_even</th>\n",
       "      <td>4000000</td>\n",
       "      <td>3999998</td>\n",
       "      <td>3999988</td>\n",
       "      <td>3999904</td>\n",
       "      <td>3999770</td>\n",
       "      <td>3398764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>atcc_staggered</th>\n",
       "      <td>4000000</td>\n",
       "      <td>3999998</td>\n",
       "      <td>3999972</td>\n",
       "      <td>3999972</td>\n",
       "      <td>3997396</td>\n",
       "      <td>3349485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RL_S001</th>\n",
       "      <td>99796358</td>\n",
       "      <td>99787104</td>\n",
       "      <td>99787100</td>\n",
       "      <td>99437858</td>\n",
       "      <td>99405860</td>\n",
       "      <td>82429052</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RM2_S001</th>\n",
       "      <td>99837678</td>\n",
       "      <td>99812266</td>\n",
       "      <td>99812230</td>\n",
       "      <td>99464896</td>\n",
       "      <td>99453182</td>\n",
       "      <td>77120530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RM2_S002</th>\n",
       "      <td>99787568</td>\n",
       "      <td>99775116</td>\n",
       "      <td>99775008</td>\n",
       "      <td>99425696</td>\n",
       "      <td>99412402</td>\n",
       "      <td>82141477</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UnAmbiguouslyMapped_ds.7</th>\n",
       "      <td>5727654</td>\n",
       "      <td>5727422</td>\n",
       "      <td>5727422</td>\n",
       "      <td>5727368</td>\n",
       "      <td>4894682</td>\n",
       "      <td>4884237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UnAmbiguouslyMapped_ds.buccal</th>\n",
       "      <td>600000</td>\n",
       "      <td>599955</td>\n",
       "      <td>599955</td>\n",
       "      <td>599950</td>\n",
       "      <td>575314</td>\n",
       "      <td>574928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UnAmbiguouslyMapped_ds.cityparks</th>\n",
       "      <td>1200000</td>\n",
       "      <td>1199969</td>\n",
       "      <td>1199969</td>\n",
       "      <td>1199956</td>\n",
       "      <td>1186747</td>\n",
       "      <td>1182428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UnAmbiguouslyMapped_ds.gut</th>\n",
       "      <td>500000</td>\n",
       "      <td>499994</td>\n",
       "      <td>499994</td>\n",
       "      <td>499993</td>\n",
       "      <td>487733</td>\n",
       "      <td>486953</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UnAmbiguouslyMapped_ds.hous1</th>\n",
       "      <td>750000</td>\n",
       "      <td>749973</td>\n",
       "      <td>749973</td>\n",
       "      <td>749964</td>\n",
       "      <td>736381</td>\n",
       "      <td>735741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UnAmbiguouslyMapped_ds.hous2</th>\n",
       "      <td>500000</td>\n",
       "      <td>499991</td>\n",
       "      <td>499991</td>\n",
       "      <td>499990</td>\n",
       "      <td>490350</td>\n",
       "      <td>489966</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UnAmbiguouslyMapped_ds.nycsm</th>\n",
       "      <td>500000</td>\n",
       "      <td>499991</td>\n",
       "      <td>499991</td>\n",
       "      <td>499988</td>\n",
       "      <td>489270</td>\n",
       "      <td>488914</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UnAmbiguouslyMapped_ds.soil</th>\n",
       "      <td>2500000</td>\n",
       "      <td>2499923</td>\n",
       "      <td>2499923</td>\n",
       "      <td>2499906</td>\n",
       "      <td>2466909</td>\n",
       "      <td>2457901</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  total_reads  reads_after_star  \\\n",
       "sample_name                                                       \n",
       "atcc_even                             4000000           3999998   \n",
       "atcc_staggered                        4000000           3999998   \n",
       "RL_S001                              99796358          99787104   \n",
       "RM2_S001                             99837678          99812266   \n",
       "RM2_S002                             99787568          99775116   \n",
       "UnAmbiguouslyMapped_ds.7              5727654           5727422   \n",
       "UnAmbiguouslyMapped_ds.buccal          600000            599955   \n",
       "UnAmbiguouslyMapped_ds.cityparks      1200000           1199969   \n",
       "UnAmbiguouslyMapped_ds.gut             500000            499994   \n",
       "UnAmbiguouslyMapped_ds.hous1           750000            749973   \n",
       "UnAmbiguouslyMapped_ds.hous2           500000            499991   \n",
       "UnAmbiguouslyMapped_ds.nycsm           500000            499991   \n",
       "UnAmbiguouslyMapped_ds.soil           2500000           2499923   \n",
       "\n",
       "                                  reads_after_trimmomatic  \\\n",
       "sample_name                                                 \n",
       "atcc_even                                         3999988   \n",
       "atcc_staggered                                    3999972   \n",
       "RL_S001                                          99787100   \n",
       "RM2_S001                                         99812230   \n",
       "RM2_S002                                         99775008   \n",
       "UnAmbiguouslyMapped_ds.7                          5727422   \n",
       "UnAmbiguouslyMapped_ds.buccal                      599955   \n",
       "UnAmbiguouslyMapped_ds.cityparks                  1199969   \n",
       "UnAmbiguouslyMapped_ds.gut                         499994   \n",
       "UnAmbiguouslyMapped_ds.hous1                       749973   \n",
       "UnAmbiguouslyMapped_ds.hous2                       499991   \n",
       "UnAmbiguouslyMapped_ds.nycsm                       499991   \n",
       "UnAmbiguouslyMapped_ds.soil                       2499923   \n",
       "\n",
       "                                  reads_after_priceseq  reads_after_cdhitdup  \\\n",
       "sample_name                                                                    \n",
       "atcc_even                                      3999904               3999770   \n",
       "atcc_staggered                                 3999972               3997396   \n",
       "RL_S001                                       99437858              99405860   \n",
       "RM2_S001                                      99464896              99453182   \n",
       "RM2_S002                                      99425696              99412402   \n",
       "UnAmbiguouslyMapped_ds.7                       5727368               4894682   \n",
       "UnAmbiguouslyMapped_ds.buccal                   599950                575314   \n",
       "UnAmbiguouslyMapped_ds.cityparks               1199956               1186747   \n",
       "UnAmbiguouslyMapped_ds.gut                      499993                487733   \n",
       "UnAmbiguouslyMapped_ds.hous1                    749964                736381   \n",
       "UnAmbiguouslyMapped_ds.hous2                    499990                490350   \n",
       "UnAmbiguouslyMapped_ds.nycsm                    499988                489270   \n",
       "UnAmbiguouslyMapped_ds.soil                    2499906               2466909   \n",
       "\n",
       "                                  nonhost_reads  \n",
       "sample_name                                      \n",
       "atcc_even                               3398764  \n",
       "atcc_staggered                          3349485  \n",
       "RL_S001                                82429052  \n",
       "RM2_S001                               77120530  \n",
       "RM2_S002                               82141477  \n",
       "UnAmbiguouslyMapped_ds.7                4884237  \n",
       "UnAmbiguouslyMapped_ds.buccal            574928  \n",
       "UnAmbiguouslyMapped_ds.cityparks        1182428  \n",
       "UnAmbiguouslyMapped_ds.gut               486953  \n",
       "UnAmbiguouslyMapped_ds.hous1             735741  \n",
       "UnAmbiguouslyMapped_ds.hous2             489966  \n",
       "UnAmbiguouslyMapped_ds.nycsm             488914  \n",
       "UnAmbiguouslyMapped_ds.soil             2457901  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Q. What percentage of the simulated reads are removed by the IDseq QC steps?\n",
    "\n",
    "sample_table = pd.read_csv('./data/idseq/project-cell_reports_benchmarking_metagenomics_tools_sample-table_v3.13.csv')\n",
    "\n",
    "sample_table.index = sample_table['sample_name']\n",
    "print(sample_table['nonhost_reads'] / sample_table['total_reads'])\n",
    "\n",
    "sample_table['host_reads'] = sample_table['total_reads'] - sample_table['nonhost_reads']\n",
    "#sample_table[['host_reads','nonhost_reads']].plot(kind='bar',stacked=True)\n",
    "\n",
    "sample_table[['total_reads','reads_after_star','reads_after_trimmomatic','reads_after_priceseq','reads_after_cdhitdup','nonhost_reads']]\n",
    "\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
